bin/dicomwebjs.js

```diff
@@ -1,35 +1,115 @@
 #!/usr/bin/env node
-import { Command } from 'commander';
-import { dicomweb, instanceDicom, dumpDicom } from '../src/index.js';
+import fs from "fs";
+import path from "path";
+import { Command } from "commander"
+import { dicomweb, instanceDicom, dumpDicom, utils } from "../src/index.js"
 
-const program = new Command();
+const program = new Command()
+
+program.option(
+  "-s, --study <studyInstanceUID>",
+  "Download a specific study instance UID"
+)
+
+program
+  .name("dicomwebjs")
+  .description("dicomwebjs based tools for manipulation of DICOMweb")
+  .version("0.0.1")
+  .option("--seriesUID <seriesUID>", "For a specific seriesUID")
 
 program
-  .name('dicomwebjs')
-  .description('dicomwebjs based tools for manipulation of DICOMweb')
-  .version('0.0.1')
-  .option('--seriesUID <seriesUID>', 'For a specific seriesUID');
-
-program.command('dump')
-  .description('Dump a dicomweb file')
-  .argument('<dicomwebUrl>', 'dicomweb URL or file location')
+  .command("dump")
+  .description("Dump a dicomweb file")
+  .argument("<dicomwebUrl>", "dicomweb URL or file location")
+  .option("--debug", "Set debug level logging")
   .action(async (fileName, options) => {
-    const qido = await dicomweb.readDicomWeb(fileName, options);
+    const qido = await dicomweb.readDicomWeb(fileName, options)
     for (const dict of qido) {
-      dumpDicom({ dict });
+      dumpDicom({ dict })
+    }
+  })
+
+program
+  .command("download")
+  .description("Download dicomweb file(s)")
+  .argument("<dicomwebUrl>", "dicomweb URL to the base DICOMweb service")
+  .option(
+    "-d, --directory <dicomwebdir>",
+    "Download to local DICOMweb directory"
+  )
+  .option("--debug", "Set debug level logging")
+  .action(async (fileName, options) => {
+    utils.logger.setOptions(options)
+    let downloadUrls = [fileName]
+    if (options.study) {
+      downloadUrls = await dicomweb.queryDownloads(fileName, options)
+    }
+    for (const downloadUrl of downloadUrls) {
+      const data = await dicomweb.readDicomWeb(downloadUrl, options)
+      dicomweb.store(downloadUrl, data, options)
+    }
+  })
+
+program
```

Comment:
Can you extract the individual program arguments to individual files?  Something like bin/cliDownload   I think you did that in another PR for me, and I liked how you did that there.